libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts

```diff
@@ -0,0 +1,276 @@
+import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import { Moment } from 'moment';
+import { forkJoin, BehaviorSubject } from 'rxjs';
+import { take } from 'rxjs/operators';
+
+import { MapboxMapService } from '@tamu-gisc/maps/mapbox';
+import { View } from 'typeorm/schema-builder/view/View';
+
+interface StateRecord {
+  STATE: string;
+  pop: number;
+  cases: number;
+  deaths: number;
+  infectionRate: number;
+  deathRate: number;
+}
+
+interface CountyRecord {
+  fips: string;
+  pop: number;
+  cases: number;
+  deaths: number;
+  infectionRate: number;
+  deathRate: number;
+}
+
+interface IInfoBox {
+  name: string;
+  pop: number;
+  cases: number;
+  deaths: number;
+  infectionRate: number;
+  deathRate: number;
+}
```

Comment:
You can take this a step further and make a common interface which these extend.

`pop`, `cases`, `deaths`, `infectionRate`, and `deathRate` appear to be shared properties with the same types.

You can do this, for example:

```js
interface BaseRecord {
    pop: number;
    cases: number;
    deaths:  number;
    infectionRate: number;
    deathRate: number;
}
```

And then:

```js
interface StateRecord extends BaseRecord {
    STATE: string;
}
```
Lastly, in places where you have interfaces inside a decorated class file, please include the types/interfaces at the end. The decorator should be displayed immediately after imports.

Typescript doesn't care that interfaces or types are defined before using them.

Reply:
I'll make this fix. Also, does it matter if we declare the interfaces in a seperate ts file and import it to the component implementation or should we keep those at the bottom?

Reply:
You can definitely split it up into a different file. Normally, the data interfaces are stored alongside the data access API services/controllers, however the data for this map isn't generated by anything in the mono-repo so I'm not too sure where it would be saved. If the same interfaces will be used by the other maps you plan to add, then at that point it would make a whole lot more sense to decouple them from the component altogether.

Perhaps in:

`libs/geoservices/covid/src/lib/pages/maps/types`?